stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -231,89 +196,60 @@ extension ${Self}: BinaryFloatingPoint {
 
-  /// The raw encoding of the value's exponent field.
-  ///
-  /// This value is unadjusted by the type's exponent bias.
   @inlinable
   public var exponentBitPattern: UInt {
     return UInt(bitPattern &>> UInt${bits}(${Self}.significandBitCount)) &
       ${Self}._infinityExponent
   }
 
-  /// The raw encoding of the value's significand field.
-  ///
-  /// The `significandBitPattern` property does not include the leading
-  /// integral bit of the significand, even for types like `Float80` that
-  /// store it explicitly.
   @inlinable
   public var significandBitPattern: ${RawSignificand} {
     return ${RawSignificand}(bitPattern) & ${Self}._significandMask
   }
 
-  /// Creates a new instance from the specified sign and bit patterns.
-  ///
-  /// The values passed as `exponentBitPattern` and `significandBitPattern` are
-  /// interpreted in the binary interchange format defined by the [IEEE 754
-  /// specification][spec].
-  ///
-  /// [spec]: http://ieeexplore.ieee.org/servlet/opac?punumber=4610933
-  ///
-  /// - Parameters:
-  ///   - sign: The sign of the new value.
-  ///   - exponentBitPattern: The bit pattern to use for the exponent field of
-  ///     the new value.
-  ///   - significandBitPattern: The bit pattern to use for the significand
-  ///     field of the new value.
   @inlinable
-  public init(sign: FloatingPointSign,
-              exponentBitPattern: UInt,
-              significandBitPattern: ${RawSignificand}) {
+  public init(
+    sign: FloatingPointSign,
+    exponentBitPattern: UInt,
+    significandBitPattern: ${RawSignificand}
+  ) {
     let signShift = ${Self}.significandBitCount + ${Self}.exponentBitCount
     let sign = UInt${bits}(sign == .minus ? 1 : 0)
     let exponent = UInt${bits}(
-      exponentBitPattern & ${Self}._infinityExponent)
+      exponentBitPattern & ${Self}._infinityExponent
+    )
     let significand = UInt${bits}(
-      significandBitPattern & ${Self}._significandMask)
+      significandBitPattern & ${Self}._significandMask
+    )
     self.init(bitPattern:
       sign &<< UInt${bits}(signShift) |
       exponent &<< UInt${bits}(${Self}.significandBitCount) |
-      significand)
+      significand
+    )
   }
 
-  /// A Boolean value indicating whether the instance's representation is in
-  /// the canonical form.
-  ///
-  /// The [IEEE 754 specification][spec] defines a *canonical*, or preferred,
-  /// encoding of a floating-point value's representation. Every `Float` or
-  /// `Double` value is canonical, but noncanonical values of the `Float80`
-  /// type exist, and noncanonical values may exist for other types that
-  /// conform to the `FloatingPoint` protocol.
-  ///
-  /// [spec]: http://ieeexplore.ieee.org/servlet/opac?punumber=4610933
   @inlinable
   public var isCanonical: Bool {
+    // All Float and Double encodings are canonical in IEEE 754.
+    //
+    // On platforms that do not support subnormals, we treat them as
+    // non-canonical encodings of zero.
```

Comment:
I think we should document this in the corresponding protocol requirement in `FloatingPoint.swift`. I like LLVM's description that offers examples:

> Examples of non-canonical encodings:
>
> * x87 pseudo denormals, pseudo NaNs, pseudo Infinity, Unnormals. These are converted to a canonical representation per hardware-specific protocol.
> * Many normal decimal floating-point numbers have non-canonical alternative encodings.
> * Some machines, like GPUs or ARMv7 NEON, do not support subnormal values. These are treated as non-canonical encodings of zero[...]

Reply:
Done!